src/tr1d1um/helper_functions.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"github.com/Comcast/webpa-common/wrp"
+	"encoding/json"
+	"strings"
+	"net/http"
+	"io"
+	"errors"
+)
+
+var (
+	ErrJsonEmpty   = errors.New("JSON payload is empty")
+)
+
+//Given some wdmp data, wraps it into a wrp object, returns the resulting payload
+func WrapInWrp(wdmpData []byte) (payload []byte, err error){
+	wrpMessage := wrp.Message{Type:wrp.SimpleRequestResponseMessageType, Payload:wdmpData}
+	payload, err = json.Marshal(wrpMessage)
+	return
+}
+
+// All we care about is the payload. Method helps abstract away work done with the WDMP object
+func GetFlavorFormat(req *http.Request, attr, formValKey, sep string) (payload[]byte, err error){
+	wdmp := new(GetWDMP)
+
+	if names := strings.Split(req.FormValue(formValKey),sep); len(names) > 0 {
+		wdmp.Command = COMMAND_GET
+		wdmp.Names = names
+	} else{
+		err = errors.New("names is a required property for GET")
+		return
+	}
+
+	if attributes := req.FormValue(attr); attributes != "" {
+		wdmp.Command = COMMAND_GET_ATTRS
+	}
+
+	payload, err = json.Marshal(wdmp)
+	return
+}
+
+func SetFlavorFormat(req *http.Request, ReadEntireBody func(io.Reader)(payload []byte, err error)) (payload[]byte, err error){
+	wdmp := new(SetWDMP)
+	DecodeJsonPayload(req, wdmp, ReadEntireBody)
+
+	wdmp.Command, err = ValidateAndGetCommand(req, wdmp)
+
+	if err != nil {
+		return
+	}
+
+	payload, err = json.Marshal(wdmp)
+	return
+}
+
+func ValidateAndGetCommand(req *http.Request, wdmp *SetWDMP) (command string, err error){
+	if newCid := req.Header.Get(HEADER_WPA_SYNC_NEW_CID); newCid != "" {
+		wdmp.OldCid = req.Header.Get(HEADER_WPA_SYNC_OLD_CID)
+		wdmp.NewCid = newCid
+		wdmp.SyncCmc =  req.Header.Get(HEADER_WPA_SYNC_CMC)
+		command, err = validateSETParams(false, wdmp, COMMAND_TEST_SET)
+	} else {
+		command, err = validateSETParams(true, wdmp, "")
+	}
+	return
+}
+
+
+/*  -Inputs-:
+ **checkingForSetAttr**: true if we're checking for the required parameter properties for the SET_ATTRIBUTES command
+		These properties are: attributes and name
+
+ **wdmp**: the WDMP object from which we retrieve the parameters
+
+ **override**: overrides the final suggested command if non-empty. Useful if one just wants to check for SET command
+		parameter properties (value, dataType, name)
+
+ 	-Outputs-:
+ **command**: the final command based on the analysis of the parameters
+ **err**: it is non-nil if any required property is violated
+*/
+func validateSETParams(checkingForSetAttr bool, wdmp *SetWDMP, override string) (command string, err error){
```

Comment:
I did this method having in mind this description: "If you provide both value and attributes by default it will do set values" from [swagger](https://swagger.webpa.comcast.net/#!/webpa/set)
If this description referred to the logic that applies on the device side and not in TR1D1UM, then this was a misunderstanding. @schmidtw